src/server/templates/zod.ts

```diff
@@ -0,0 +1,323 @@
+    let type = arrayTypes.find(({ id }) => id === arg.type_id)
+    if (type) {
+        // If it's an array type, the name looks like `_int8`.
+        const elementTypeName = type.name.substring(1)
+        return `z.array(z.${basicZodType(elementTypeName)})` + (arg.has_default ? '.optional()' : '')
+    }
+    type = types.find(({ id }) => id === arg.type_id)
+    if (type) {
+        return "z." + basicZodType(type.format) + (arg.has_default ? '.optional()' : '')
+    }
+
+    console.debug(`Function: Unknown type ${arg.type_id}`)
+
+    return `z.unknown()` + (arg.has_default ? '.optional()' : '')
+}
+
+function basicZodType(pgType: string): string {
+    // Array
+    if (pgType.startsWith("_")) {
+        return basicZodType(pgType.substring(1)) + ".array()"
+    }
+
+    if ([ 'bool', 'boolean' ].includes(pgType)) {
+        return "boolean()"
+    }
+
+    if (['int2', 'int4', 'int8', 'float4', 'float8', 'numeric', 'integer', 'bigint', 'oid'].includes(pgType)) {
+        return 'number()'
+    }
+
+    if (
+        [
+            'bytea',
+            'bpchar',
+            'varchar',
+            'text',
+            'citext',
+            'uuid',
+            'vector',
+            'json',
+            'jsonb',
+            'inet',
+            'cidr',
+            'macaddr',
+            'macaddr8',
+            'character varying',
+        ].includes(pgType)
+    ) {
+        return 'string()'
+    }
+
+    if (["date", "time", "timetz", "timestamp", "timestamptz", "timestamp with time zone"].includes(pgType)) {
+        return 'date()'
+    }
+
+    console.debug(`Basic Zod Type: Unknown type ${pgType}`)
+
+    // Everything else is an enum
+    return "string()"
+}
+
+const IP_REGEX = "^((((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))|((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))))\\/[0-9]{1,3}$"
+const UUID_REGEX = "^[0-9a-fA-F]{8}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{12}$"
+
+function extractExtraZodMethods(column: PostgresColumn): string[] {
+    const methods: string[] = []
+
+    // UUID
+    if (column.format === "uuid") {
+        methods.push(`regex(/${UUID_REGEX}/)`)
+
+        if (column.default_value === "gen_random_uuid()") {
+            methods.push("default(uuidv4)")
```

Comment:
lets not add default values to the zod schema -> they are added in the BE anyways.

Reply:
otherwise you might have a case where someone queries the backend without the `id` column, passes it through the zod schema and then has a wrong `id` set.